das/state.go

```diff
@@ -1,53 +1,140 @@
 package das
 
-import (
-	"sync"
-	"time"
-)
-
-// state collects information about the DASer process. Currently, there are
-// only two sampling routines: the main sampling routine which performs sampling
-// over current network headers, and the `catchUp` routine which performs sampling
-// over past headers from the last sampled checkpoint.
+// represents current state of sampling
 type state struct {
-	sampleLk sync.RWMutex
-	sample   RoutineState // tracks information related to the main sampling routine
+	priority   []uint64        // list of headers heights that will be sampled with higher priority
+	inProgress map[uint64]bool // keeps track of inProgress items with priority flag stored as value
+	failed     map[uint64]int  // stores heights of failed headers with amount of attempt as value
 
-	catchUpLk sync.RWMutex
-	catchUp   JobInfo // tracks information related to the `catchUp` routine
+	priorityBusy bool // semaphore to allow only one priority item to be processed at time
```

Comment:
We don't need to implement the solution for broken Mamaki issues. Details are dicussed in sync